module-workers.tf

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  worker_count_expected = coalesce(one(module.workers[*].worker_count_expected), 0)
+
+  # Distinct list of compartments for enabled worker pools
+  worker_compartments = distinct(compact([
```

Comment:
Using different compartments make things more difficult in my view: permissions, debugging in the console. I know there are users who roll this way, especially because they want to restrict access to compute or they want to cost by compartment. 

In those cases, if users want isolation for this kind of purposes, it is better they stand up a different cluster.

Reply:
I agree with you, but I know it's being done in some cases.